BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -184,9 +192,26 @@ extension AttributedLabel {
             urlHandler = environment.urlHandler
             layoutDirection = environment.layoutDirection
 
-            if !isMeasuring, previousAttributedText != attributedText {
-                links = attributedLinks(in: model.attributedText) + detectedDataLinks(in: model.attributedText)
-                accessibilityLinks = accessibilityLinks(for: links, in: model.attributedText)
+            if !isMeasuring {
+                if previousAttributedText != attributedText {
+                    links = attributedLinks(in: model.attributedText) + detectedDataLinks(in: model.attributedText)
+                    accessibilityLinks = accessibilityLinks(for: links, in: model.attributedText)
+                }
+
+                if let shadow = model.shadow {
+                    layer.shadowRadius = shadow.radius
+                    layer.shadowOpacity = Float(shadow.opacity)
+                    layer.shadowOffset = CGSize(width: shadow.offset.horizontal, height: shadow.offset.vertical)
+                    layer.shadowColor = shadow.color.cgColor
+
+                    // enable rasterization on this layer, since text is too complex for a shadow path
```

Comment:
I take it this is faster at rendering than turning the text into a path and setting that as the shadow path? Or is that what this comment means?

Reply:
Yep, although I did not test it myself, all the reading material I found on layer shadows suggested offscreen compositing for cases where a path is impractical ([this, for example](https://swiftlemma.com/2017/08/20/better-shadow-performance-on-views/)). I can rewrite this comment to clarify.